geode-management/src/main/java/org/apache/geode/management/configuration/Region.java

```diff
@@ -213,4 +224,88 @@ public void setAction(ExpirationAction action) {
       this.action = action;
     }
   }
+
+  public enum EvictionType {
+    ENTRY_COUNT,
+    MEMORY_SIZE,
+    HEAP_PERCENTAGE
+  }
+
+  public enum EvictionAction {
+    LOCAL_DESTROY,
+    OVERFLOW_TO_DISK
+  }
+
+  public static class Eviction implements Serializable {
+    private EvictionType type;
+    private EvictionAction action;
+    private Integer limit;
+    private String objectSizer;
+
+    public Eviction() {}
+
+    public Eviction(EvictionType type, EvictionAction action, Integer limit, String objectSizer) {
+      this.type = type;
+      this.action = action;
+      this.limit = limit;
+      this.objectSizer = objectSizer;
+    }
+
+    public EvictionType getType() {
+      return type;
+    }
+
+    public void setType(EvictionType type) {
+      if (this.type != null && this.type != type) {
+        throw new IllegalArgumentException("Type conflict. Type is already set to " + this.type);
+      }
+      this.type = type;
+    }
+
+    public EvictionAction getAction() {
+      return action;
+    }
+
+    public void setAction(EvictionAction action) {
+      this.action = action;
+    }
+
+    public Integer getEntryCount() {
+      if (type == EvictionType.ENTRY_COUNT) {
+        return limit;
+      }
+      return null;
+    }
+
+    public void setEntryCount(Integer entryCount) {
+      if (type != null && type != EvictionType.ENTRY_COUNT) {
+        throw new IllegalArgumentException("Type conflict. Type is already set to " + type);
+      }
+      type = EvictionType.ENTRY_COUNT;
+      this.limit = entryCount;
+    }
+
+    public Integer getMemorySizeMb() {
+      if (type == EvictionType.MEMORY_SIZE) {
+        return limit;
+      }
+      return null;
+    }
+
+    public void setMemorySizeMb(Integer memorySizeMb) {
```

Comment:
add javadocs describing when it will throw
also describe that it will also set the type to...l